softlayer/resource_softlayer_file_storage.go

```diff
@@ -594,18 +594,20 @@ func WaitForStorageAvailable(d *schema.ResourceData, meta interface{}) (interfac
 
 			// Check volume status.
 			log.Println("Checking volume status.")
-			resultStr := ""
+			var resultObj interface{}
 			err = sess.DoRequest(
 				"SoftLayer_Network_Storage",
 				"getObject",
 				nil,
 				&sl.Options{Id: &id, Mask: "volumeStatus"},
-				&resultStr,
+				&resultObj,
 			)
```

Comment:
@minsikl I saw that a problem existed when using XMLRPC, since the classic form object mask (used by default above) was not supported in our xmlrpc implementation and that is the only mask form that works in this case.

I have now fixed this in softlayer-go, so the code above will work whether using REST or XMLRPC. So we should simplify it in the provider now and update softlayer-go in the vendor subdirectory again.